### ISSUE TITLE
Error in plugin solidity-coverage: TypeError: cur._node._vm.on is not a function

### ISSUE THREAD

**Issue by wwHysenberg** [sc-forks/solidity-coverage#737]
Using solidity 0.8.9 in a standard hardhat environment

```
$ npx hardhat coverage

Version
=======
> solidity-coverage: v0.7.21

Instrumenting for coverage...
=============================

> interfaces/tokens/IWETH9.sol
> tokens/WETH9.sol

Compilation:
============

Compiled 6 Solidity files successfully
Creating Typechain artifacts in directory typechain for target ethers-v5
Successfully generated Typechain artifacts!
Error in plugin solidity-coverage: TypeError: cur._node._vm.on is not a function

HardhatPluginError: TypeError: cur._node._vm.on is not a function
    at SimpleTaskDefinition.action (/path/to/project/node_modules/solidity-coverage/plugins/hardhat.plugin.js:243:35)
    at async Environment._runTaskDefinition (/path/to/project/node_modules/hardhat/src/internal/core/runtime-environment.ts:308:14)
    at async Environment.run (/path/to/project/node_modules/hardhat/src/internal/core/runtime-environment.ts:156:14)
    at async main (/path/to/project/node_modules/hardhat/src/internal/cli/cli.ts:269:7)
```

**Comment by pancudaniel7** [comment]
I have the same problem:
<img width="975" alt="Screenshot 2022-09-03 at 14 58 14" src="https://user-images.githubusercontent.com/9675142/188269435-e8825b25-615a-4af5-b7e6-71518cf02662.png">

**Comment by saumya66** [comment]
Facing the same error 👇👇

![image](https://user-images.githubusercontent.com/60464414/188280674-9a13c396-08fe-4513-8bc9-61fcd0ad56ee.png)

**Comment by f-hk** [comment]
i have same error too!
![image](https://user-images.githubusercontent.com/107092758/188322728-547afe9a-9bb8-4783-9979-81aa709ff0a8.png)

**Comment by 0xslipk** [comment]
I have the same error with a lower version, using pnpm instead of yarn:
<img width="705" alt="image" src="https://user-images.githubusercontent.com/3802516/188322790-31311d3a-bc4b-4aa1-abba-76609f559215.png">

**Comment by 0xslipk** [comment]
@wwHysenberg  looks like the problem is the newest version of [hardhat@2.11.0](https://www.npmjs.com/package/hardhat/v/2.11.0) with [hardhat@2.10.2](https://www.npmjs.com/package/hardhat/v/2.10.2) work as before.

**Comment by wwHysenberg** [comment]
> @wwHysenberg looks like the problem is the newest version of [hardhat@2.11.0](https://www.npmjs.com/package/hardhat/v/2.11.0) with [hardhat@2.10.2](https://www.npmjs.com/package/hardhat/v/2.10.2) work as before.

can confirm, coverage works after setting `"hardhat": "=2.10.2"`. Thanks!

**Comment by alcuadrado** [comment]
Hey @cgewecke! I'm happy to jump into this. Can you point me to the right branch?

**Comment by mkaya13** [comment]
`Error in plugin solidity-coverage: TypeError: cur._node._vm.on is not a function`

I have the same issue unfortunately.
My package.json is shown in below:
`{
  "name": "from-stract-hardhat-fund-me-fcc",
  "version": "1.0.0",
  "main": "index.js",
  "license":"MIT",
  "devDependencies": {
    "@chainlink/contracts": "^0.4.1",
    "@ethersproject/abi": "^5.4.7",
    "@ethersproject/providers": "^5.4.7",
    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
    "@nomicfoundation/hardhat-toolbox": "^1.0.1",
    "@nomiclabs/hardhat-ethers": "npm:hardhat-deploy-ethers",
    "@nomiclabs/hardhat-etherscan": "^3.0.0",
    "@nomiclabs/hardhat-waffle": "^2.0.3",
    "@typechain/ethers-v5": "^10.1.0",
    "@typechain/hardhat": "^6.1.2",
    "chai": "^4.2.0",
    "dotenv": "^16.0.2",
    "ethers": "^5.7.0",
    "hardhat": "^2.11.0",
    "hardhat-deploy": "^0.11.13",
    "hardhat-gas-reporter": "^1.0.8",
    "solhint": "^3.3.7",
    "solidity-coverage": "^0.7.21",
    "typechain": "^8.1.0"
  },
  "dependencies": {}
}
`

**Comment by mkaya13** [comment]
I have the same issue :
`Error in plugin solidity-coverage: TypeError: cur._node._vm.on is not a function`

My package.json as follows:
{
  "name": "from-stract-hardhat-fund-me-fcc",
  "version": "1.0.0",
  "main": "index.js",
  "license":"MIT",
  "devDependencies": {
    "@chainlink/contracts": "^0.4.1",
    "@ethersproject/abi": "^5.4.7",
    "@ethersproject/providers": "^5.4.7",
    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
    "@nomicfoundation/hardhat-toolbox": "^1.0.1",
    "@nomiclabs/hardhat-ethers": "npm:hardhat-deploy-ethers",
    "@nomiclabs/hardhat-etherscan": "^3.0.0",
    "@nomiclabs/hardhat-waffle": "^2.0.3",
    "@typechain/ethers-v5": "^10.1.0",
    "@typechain/hardhat": "^6.1.2",
    "chai": "^4.2.0",
    "dotenv": "^16.0.2",
    "ethers": "^5.7.0",
    "hardhat": "^2.11.0",
    "hardhat-deploy": "^0.11.13",
    "hardhat-gas-reporter": "^1.0.8",
    "solhint": "^3.3.7",
    "solidity-coverage": "^0.7.21",
    "typechain": "^8.1.0"
  },
  "dependencies": {}
}

**Comment by cgewecke** [comment]
Hotfix pending via https://github.com/sc-forks/solidity-coverage/pull/740

**Comment by cgewecke** [comment]
This is fixed in `solidity-coverage@latest` (v0.8.0)

**Comment by mattjaf** [comment]
run `yarn hardhat clean` might be due to stale artifacts
